base/multidimensional.jl

```diff
@@ -207,11 +207,11 @@ function _unsafe_getindex(::LinearIndexing, src::AbstractArray, I::AbstractArray
     D = eachindex(dest)
     Ds = start(D)
     s = 0
-    for b in eachindex(I)
+    @inbounds for b in eachindex(I)
```

Comment:
Sure, but we've not annotated next with `@boundscheck/@inbounds` or `@_propagate_inbounds_meta` yet. It may be something we want to do — especially since it'd allow us to actually follow the iteration protocol in these performance sensitive cases.

In any case, all five cases of expanded scope (there's also the `@inbounds @nloops`) here are iteration over AbstractArrays or their eachindex iterators or an infinite `repeated` iterator, and we're already assuming that AbstractArrays must be implemented correctly for this to work.